tests/jballerina-unit-test/src/test/resources/test-src/annotations/external_dependency_annotation.bal

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+// Functions
+@java:ExternalDependency
+function foo(int x) returns int {
+    return x * 2;
+}
+
+// Classes
+@java:ExternalDependency
+class BazClass {
+    int x = 0;
+    string y = "";
+}
+
+// Type Definitions
+@java:ExternalDependency
+type FooRec record {
+    int x;
+    string y;
+};
```

Comment:
Shall we add two test case.

1. Add the annotation for a remote function inside a service/class level method
2. Add the annotation for a function that already have a anotation.

```
@java:ExternalDependency
@test:Config
function test2(*A aa) {
    
}
```

Reply:
We had tests like that before. But removed them because, as @MaryamZi pointed out, such test cases are for validating the correctness of the compiler, rather than the annotation itself. Therefore, we removed them and added a test case for validating the `AttachPoints` of the `AnnotationSymbol`.